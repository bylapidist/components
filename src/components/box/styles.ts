import { Styles } from '@lapidist/styles';
import { BoxProps } from './index';

export const boxStyles = (props: BoxProps): Styles => props.styles || {};
