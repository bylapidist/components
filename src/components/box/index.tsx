import React, { PropsWithChildren } from 'react';
import styled from 'styled-components';
import { Styles, withStyles } from '@lapidist/styles';

export interface BoxProps extends PropsWithChildren<unknown> {
    readonly styles?: Styles;
    readonly as?: keyof JSX.IntrinsicElements;
}

export const Box: React.FC<BoxProps> = styled.div<BoxProps>`
    ${({ theme, styles }): string => withStyles(theme, styles)}
`;

Box.displayName = 'Box';
