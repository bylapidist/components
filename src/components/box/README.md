The Box component is a layout primitive to add margin, padding and colors to content.

```jsx harmony
import { ThemeProvider, Box, Text } from '@lapidist/components';

<ThemeProvider>
    <Box
        padding={4}
        backgroundColor={{ colorGroup: 'primary', colorShade: 'base' }}
    >
        <Text textColor={{ colorGroup: 'base', colorShade: 'white' }}>Hello world!</Text>
    </Box>
</ThemeProvider>
```
