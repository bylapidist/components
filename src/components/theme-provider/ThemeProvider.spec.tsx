import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';

import { ThemeProvider } from './index';
import { Box } from '../box';

test('it works', () => {
    const { container } = render(
        <ThemeProvider>
            <Box styles={{ margin: '2' }}>Hello world!</Box>
            <Box styles={{ margin: '3' }}>Hello world!</Box>
        </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
});

test('it works with breakpoints', () => {
    const { container } = render(
        <ThemeProvider>
            <Box
                styles={{
                    margin: '2',
                    breakpoints: { xl: { margin: '3' } }
                }}
            >
                Hello world!
            </Box>
        </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
});

test('it works with theme', () => {
    const { container } = render(
        <ThemeProvider theme={{ sizes: { '2': '20px' } }}>
            <Box styles={{ margin: '2' }}>Hello world!</Box>
            <Box styles={{ margin: '3' }}>Hello world!</Box>
        </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
});
