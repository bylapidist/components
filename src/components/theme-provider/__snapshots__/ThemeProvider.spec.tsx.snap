// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  margin: 0.5rem;
}

<div
  class="c0"
>
  Hello world!
</div>
`;

exports[`it works with breakpoints 1`] = `
.c0 {
  margin: 0.5rem;
}

@media only screen and (min-width:1280px) {
  .c0 {
    margin: 0.75rem;
  }
}

<div
  class="c0"
>
  Hello world!
</div>
`;

exports[`it works with theme 1`] = `
.c0 {
  margin: 20px;
}

<div
  class="c0"
>
  Hello world!
</div>
`;
